File: coding-practice/50 Coding Challenges part 1/Python/problem16.py
# create a function that will return a Boolean specifying if a number is prime
import math
def is_prime(x):
    if x<2:
        return False
    elif x==2:
        return True
    # a number is considered to be prime if the modulus of that number with its square root is not equal to zero
    # sebuah bilangan dikatakan prima jika hasil bagi antara bilangan tersebut dengan pembulatan dari akarnya 
    # tidak sama dengan nol
    for i in range(2, int(math.sqrt(x) + 1)):
        if x%i == 0:
            return False
    return True
x = int(input('insert a number '))
print("{} is prime?: {}".format(x, str(is_prime(x))))

File: coding-practice/50 Coding Challenges part 1/C#/problem16.cs
// coding challenge 16 : create a function that will return a boolean specifying if a number is prime
using System;
class problem16
{
    static bool is_prime(int number){
        if (number<2){
            return false;
        }

        else if(number==2){
            return true;
        }

        int max_div = (int)Math.Sqrt(number);
        for(int i=2; i<=max_div; i++){
            if(number%i == 0){
                return false;
            }
        }
        return true;

    }

    static void Main(string[] args){
        int[] numbers = {1, 8, 3, 4, 19, 21, 23, 45};
        for(int i=0; i<numbers.Length; i++){
            Console.WriteLine("Is "+numbers[i]+" prime? : "+is_prime(numbers[i]));
        }
    }
}